net/sched/sch_dsprio.c

```diff
@@ -234,9 +250,10 @@ static int dsprio_init(struct Qdisc *sch, struct nlattr *opt,
 	unsigned int min_limit = 1;
 
 	/* Initialise all queues, one for each possible priority. */
-	for (prio = 0; prio < DSPRIO_MAX_PRIORITY; prio++)
+	for (prio = 0; prio < DSPRIO_MAX_PRIORITY; prio++) {
 		__skb_queue_head_init(&q->qdiscs[prio]);
-
+		memset(&q->qstats[prio], 0, sizeof(q->qstats[prio]));
```

Comment:
You can put this `memset()` outside of the loop and apply it to the whole array.